pkg/skaffold/config/types_test.go

```diff
@@ -72,3 +73,84 @@ func TestStringOrUndefined(t *testing.T) {
 		})
 	}
 }
+
+func TestMuted(t *testing.T) {
+	tests := []struct {
+		phases                  []string
+		expectedMuteBuild       bool
+		expectedMuteTest        bool
+		expectedMuteStatusCheck bool
+		expectedMuteDeploy      bool
+	}{
+		{
+			phases:                  nil,
+			expectedMuteBuild:       false,
+			expectedMuteTest:        false,
+			expectedMuteStatusCheck: false,
+			expectedMuteDeploy:      false,
+		},
+		{
+			phases:                  []string{"build"},
+			expectedMuteBuild:       true,
+			expectedMuteTest:        false,
+			expectedMuteStatusCheck: false,
+			expectedMuteDeploy:      false,
+		},
+		{
+			phases:                  []string{"test"},
+			expectedMuteBuild:       false,
+			expectedMuteTest:        true,
+			expectedMuteStatusCheck: false,
+			expectedMuteDeploy:      false,
+		},
+		{
+			phases:                  []string{"status-check"},
+			expectedMuteBuild:       false,
+			expectedMuteTest:        false,
+			expectedMuteStatusCheck: true,
+			expectedMuteDeploy:      false,
+		},
+		{
+			phases:                  []string{"deploy"},
+			expectedMuteBuild:       false,
+			expectedMuteTest:        false,
+			expectedMuteStatusCheck: false,
+			expectedMuteDeploy:      true,
+		},
+		{
+			phases:                  []string{"build", "test", "status-check", "deploy"},
+			expectedMuteBuild:       true,
+			expectedMuteTest:        true,
+			expectedMuteStatusCheck: true,
+			expectedMuteDeploy:      true,
+		},
+		{
+			phases:                  []string{"all"},
+			expectedMuteBuild:       true,
+			expectedMuteTest:        true,
+			expectedMuteStatusCheck: true,
+			expectedMuteDeploy:      true,
+		},
+		{
+			phases:                  []string{"none"},
```

Comment:
What should happen if there's `none,build`? 😱

Reply:
no idea :-) This code is very lenient for sure